src/foundry/client/pixi/webgl/shaders/filters/transition.d.mts

```diff
@@ -0,0 +1,79 @@
+import type { FilterSystem, RenderTexture, CLEAR_MODES } from "pixi.js";
+
+export {};
+
+declare global {
+  /**
+   * A filter specialized for transition effects between a source object and a target texture.
+   */
+  class TextureTransitionFilter extends AbstractBaseFilter {
+    /**
+     * If this filter requires padding (according to type)
+     */
+    #requirePadding: boolean;
+
+    /**
+     * Transition types for this shader.
+     */
+    static get TYPES(): Record<string, string>;
+
+    /**
+     * The transition type (see {@link TextureTransitionFilter.TYPES}).
+     * @defaultValue TextureTransitionFilter.TYPES.FADE
+     */
+    get type(): string;
+
+    set type(type: string);
+
+    /**
+     * Sampler target for this filter.
+     */
+    set targetTexture(targetTexture: PIXI.Texture);
+
+    /**
+     * Animate a transition from a subject SpriteMesh/PIXI.Sprite to a given texture.
+     * @param subject - The source mesh/sprite to apply a transition.
+     * @param texture - The target texture.
+     * @param options - Animation options.
+     * @returns   A Promise which resolves to true once the animation has concluded
+     *            or false if the animation was prematurely terminated
+     */
+    static animate(
+      subject: PIXI.Sprite | SpriteMesh,
+      texture: PIXI.Texture,
+      options?: {
+        /**
+         * The transition type
+         * @defaultValue `TYPES.FADE`
+         */
+        type?: typeof TextureTransitionFilter.TYPES;
+        /**
+         * The name of the {@link CanvasAnimation}.
+         */
+        name?: string | symbol;
+        /**
+         * The animation duration
+         * @defaultValue 1000
+         */
+        duration?: number;
+        /**
+         * The easing function of the animation
+         */
+        easing?: Function | string;
```

Comment:
`easing` seems to be passed to `CanvasAnimation.animate` which passes it to `CanvasAnimation.#animateFrame`:
```js
  static #animateFrame(deltaTime, animation) {
    const {attributes, duration, ontick} = animation;

    // Compute animation timing and progress
    const dt = this.ticker.elapsedMS;     // Delta time in MS
    animation.time += dt;                 // Total time which has elapsed
    const complete = animation.time >= duration;
    const pt = complete ? 1 : animation.time / duration; // Proportion of total duration
    const pa = animation.easing ? animation.easing(pt) : pt;
```

This indicates it's only ever called with a number and always expects a number. Therefore `((percentage: number) => number) | string` here seems like the best type.

---

Alternatively if any function is truly desired please use `AnyFunction` instead.

`Function` is a deeply misleading name because it also allows classes:
```ts
const x: Function = class {}
```

This is more JS's fault than anything `(class {}) instanceof Function` returns true in JS.

Reply:
`CanvasAnimation.animate` does this:
```js
if ( typeof easing === "string" ) easing = this[easing];
```
and that seems to be for getting one of the three provided functions: 
```js
static easeInOutCosine(pt) 
static easeOutCircle(pt)
static easeInCircle(pt)
```
which then gets passed along inside an `animation` object to `.#animateFrame`:
```js
const pa = animation.easing ? animation.easing(pt) : pt;
```
So custom easing functions *are* supported and should be allowed. The signature seems to consistently be `(pt: number) => number`, so should this be typed as that `| string`?